### DISCUSSION TITLE
Error encountered while using liionpack

### DISCUSSION THREAD

**Discussion by xcblovecy** [pybamm-team/liionpack#257] [Q&A] [answered]
The following error occurs when liionpack is used to simulate battery packs：
```
”RuntimeError: .../casadi/core/slice.cpp:80: Assertion "stop<=len" failed:
Slice (start=319, stop=330, step=1) out of bounds with supplied length of 320”
```
Unfortunately, this error occurs in most examples of liionpack.

pybamm：23.4.1
liionpack：0.3.5.

**Comment by TomTranter** [comment, marked as answer]
Hi, this appears to be a change in the casadi integrator from their latest package. Some code changes made in PyBaMM to address this but not made in liionpack yet. I will raise an issue on the liionpack github to address it ASAP
